File: backend/src/main/java/com/autonomofinancas/service/UsuarioService.java
package com.autonomofinancas.service;

import com.autonomofinancas.dto.request.AlterarSenhaRequest;
import com.autonomofinancas.dto.request.AtualizarUsuarioRequest;
import com.autonomofinancas.dto.request.CriarUsuarioRequest;
import com.autonomofinancas.dto.response.UsuarioPerfilResponse;
import com.autonomofinancas.dto.response.UsuarioResponse;

public interface UsuarioService {

    UsuarioResponse criar(CriarUsuarioRequest request);

    UsuarioPerfilResponse buscarPerfil();

    UsuarioPerfilResponse atualizarPerfil(AtualizarUsuarioRequest request);

    void alterarSenha(AlterarSenhaRequest request);
    
}

File: frontend/src/app/core/services/usuario.service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { environment } from '../../../environments/environment.development';
import { CriarUsuarioRequest } from '../models/criar-usuario-request';
import { Observable } from 'rxjs';
import { UsuarioResponse } from '../models/usuario-response';

@Injectable({
  providedIn: 'root',
})
export class UsuarioService {

  private readonly http = inject(HttpClient);
  private readonly apiUrl = `${environment.apiUrl}/usuarios`;

  criar(request: CriarUsuarioRequest): Observable<UsuarioResponse> {
    return this.http.post<UsuarioResponse>(
      this.apiUrl,
      request
    );
  }

}
